### PR TITLE
XRT OS Support for Ubuntu 24.04. Removed one element arrays in the driver code

### PR DESCRIPTION
**Pull request by ManojTakasi**
#### Problem solved by the commit
https://jira.xilinx.com/browse/VITIS-12053 XRT OS Support for Ubuntu 24.04

This is an intial PR to move to flexible arrays instead of one length array in structures. Reference: https://www.kernel.org/doc/html/latest/process/deprecated.html#zero-length-and-one-element-arrays

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Removed mmap xocl_gem_prime_mmap function as drm_gem_mmap() function  is capible of setting flags before mmap.

#### How problem was solved, alternative solutions (if any) and why they were rejected
Removed the xocl_gem_prime_mmap and changed some structures into flexible arrays as one-legth arrays are throwing call trace while using memcpy() function.

#### Risks (if any) associated the changes in the commit
low

#### What has been tested and how, request additional testing if necessary
Tested with u250 in ubuntu 22.04 and 24.04

#### Documentation impact (if any)
Yes

### CODE REVIEW

**Review by chvamshi-xilinx — CHANGES_REQUESTED** [review]
As discussed, please try to create  a macro instead of calling sizeof in lot of places.

**Comment by ManojTakasi** [comment]
Hi Vamshi, as suggested used struct_size in all the places.

**Comment by keryell** [comment]
I do not understand the reference to VLA here.
Where are the VLA in the code?

**Comment by keryell** [comment]
Just to clarify, the "flexible array" is a coding convention, like VLA () are a C99 features made optional starting with C11
https://en.wikipedia.org/wiki/Variable-length_array
```c
int n = 42;
float a[n];
```
It is not in C++ because you cannot have `sizeof(a)` not being `constexpr`.

**Comment by ManojTakasi** [comment]
Hi Keryell, Thanks for your clarification. Sorry for referencing wrong link. Please check this link: https://www.kernel.org/doc/html/latest/process/deprecated.html#zero-length-and-one-element-arrays

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c [hunk 1]
```diff
@@ -327,7 +327,7 @@ int xocl_program_shell(struct xocl_dev *xdev, bool force)
 		goto failed;
 
 	userpf_info(xdev, "request mgmtpf to program prp");
-	mbret = xocl_peer_request(xdev, &mbreq, sizeof(struct xcl_mailbox_req),
+	mbret = xocl_peer_request(xdev, &mbreq, sizeof(mbreq) + sizeof(int32_t),
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c:330]: struct_size here aswell?

**Review by chvamshi-xilinx — APPROVED** [review]

**Pull request merged**